packages/api/src/external/commonwell/document/document-query.ts

```diff
@@ -81,22 +83,36 @@ type File = DownloadResult & { isNew: boolean };
  * CW for them (optional) - see `downloadDocsAndUpsertFHIR()` for the default value
  */
 export async function queryAndProcessDocuments({
-  patient,
+  patient: patientParam,
   facilityId,
+  forceQuery = false,
   forceDownload,
   ignoreDocRefOnFHIRServer,
   ignoreFhirConversionAndUpsert,
   requestId,
 }: {
   patient: Patient;
   facilityId?: string | undefined;
+  forceQuery?: boolean;
   forceDownload?: boolean;
   ignoreDocRefOnFHIRServer?: boolean;
   ignoreFhirConversionAndUpsert?: boolean;
   requestId: string;
 }): Promise<number> {
-  const { log } = Util.out(`CW queryDocuments: ${requestId} - M patient ${patient.id}`);
+  const { log } = Util.out(`CW queryDocuments: ${requestId} - M patient ${patientParam.id}`);
+
   try {
+    const patient = await getPatientWithCWData(patientParam);
+    const cwData = patient.data.externalData.COMMONWELL;
+
+    const isWaitingForEnhancedCoverage =
+      (await isEnhancedCoverageEnabledForCx(patient.cxId)) &&
+      cwData.cqLinkStatus && // we're not waiting for EC if the patient was created before cqLinkStatus was introduced
+      cwData.cqLinkStatus !== "linked";
```

Comment:
If the patient was created before EC was enabled for this cx (no cqLinkStatus) OR the patient's cqLinkStatus is not `linked`, it means we're waiting for EC and should NOT run DQ.